### PR TITLE
feat: add concat() handlebar helper (issue-260)

### PR DESCRIPTION
**Pull request by avnindersingh**

### CODE REVIEW

File: src/handlebar-helpers.js [hunk 1]
```diff
@@ -65,6 +65,12 @@ const helpers = {
     }
     const hash = crypto.createHash('sha256').update(data).digest('hex');
     return hash;
+  },
+  concat: function(arr){
+    if (Array.isArray(arr)){
+      return ''.concat(...arr)
```
Comment (alewitt2) [inline src/handlebar-helpers.js:71]:
```suggestion
      return ''.concat(...arr);
```

File: src/handlebar-helpers.js [hunk 2]
```diff
@@ -65,6 +65,12 @@ const helpers = {
     }
     const hash = crypto.createHash('sha256').update(data).digest('hex');
     return hash;
+  },
+  concat: function(arr){
```
Comment (alewitt2) [inline src/handlebar-helpers.js:69]: is there a reason you want to pass in an array as opposed to accepting an arbitrary number of parameters?
ie.
```
concat: function(...arr){
  return ''.concat(...arr)
}

HandlebarHelper.concat("Hello ", "World")
```

Comment (alewitt2) [reply src/handlebar-helpers.js:69]: or better yet, accept both
```
concat: function(...arr){
  let flat = arr.flat();
  return ''.concat(...flat); 
}
```

Comment (avnindersingh) [reply src/handlebar-helpers.js:69]: > is there a reason you want to pass in an array as opposed to accepting an arbitrary number of parameters?
> ie.
> 
> ```
> concat: function(...arr){
>   return ''.concat(...arr)
> }
> 
> HandlebarHelper.concat("Hello ", "World")
> ```

@alewitt2 considering case where more than 2 strings needs to join .... feel free to change

File: src/handlebar-helpers.js [hunk 3]
```diff
@@ -65,6 +65,12 @@ const helpers = {
...
+    if (Array.isArray(arr)){
+      return ''.concat(...arr);
+    }
+    return -1;
```
Comment (alewitt2) [inline src/handlebar-helpers.js:73]:
```suggestion
  },
  concat: function(...arr) {
    let flat = arr.flat();
    return ''.concat(...flat); 
  }
```

File: src/handlebar-helpers.js [hunk 4]
```diff
@@ -65,6 +65,11 @@ const helpers = {
     }
     const hash = crypto.createHash('sha256').update(data).digest('hex');
     return hash;
+  },
+  concat: function(...arr) {
+    let flat = arr.flat();
+    return ''.concat(...flat); 
+  }
```
Comment (alewitt2) [inline src/handlebar-helpers.js:72]:
```suggestion
```

File: test/handlebar-helpers.js [hunk 5]
```diff
@@ -330,4 +330,29 @@ describe('handlebar-helper', function () {
...
+        assert.equal("", ret, 'empty array');
+    });
+    it('should concatenate and return string when a string and a number passed in array', function () {
+        ret = HandlebarHelper.concat(["Hello ", 123])
+        assert.equal("Hello 123", ret, 'concat string is "Hello 123"');
+    });
+    it('should concatenate and return string when combination of string, number and string passed in array', function () {
+        ret = HandlebarHelper.concat(["Hello ", 123, " Hello"])
+        assert.equal("Hello 123 Hello", ret, 'concat string is "Hello 123 Hello"');
+    });
```
Comment (alewitt2) [inline test/handlebar-helpers.js:357]:
```suggestion
    //--- concat() tests
    it('should return concatenate string when array has only 1 string', function () {
        ret = HandlebarHelper.concat(['Hello'])
        assert.equal('Hello', ret, 'concat string is "Hello"');
    });
    it('should return concatenate string when array has 2 strings', function () {
        ret = HandlebarHelper.concat(['Hello ','World'])
        assert.equal('Hello World', ret, 'concat string is "Hello World"');
    });
    it('should return concatenate string when array has 3 strings', function () {
        ret = HandlebarHelper.concat(['Hello ','World ','!!'])
        assert.equal('Hello World !!', ret, 'concat string is "Hello World !!"');
    });
    it('should not throw error when 0 inputs are passed in array', function () {
        ret = HandlebarHelper.concat([])
        assert.equal('', ret, 'empty array');
    });
    it('should concatenate and return string when non strings are passed in array', function () {
        ret = HandlebarHelper.concat(['Hello ', 123, ' ', true])
        assert.equal('Hello 123 true', ret, 'concat string is "Hello 123 true"');
    });
    it('should return concatenate string when params have only 1 string', function () {
        ret = HandlebarHelper.concat('Hello')
        assert.equal('Hello', ret, 'concat string is "Hello"');
    });
    it('should return concatenate string when params have 2 strings', function () {
        ret = HandlebarHelper.concat('Hello ','World')
        assert.equal('Hello World', ret, 'concat string is "Hello World"');
    });
    it('should return concatenate string when params have 3 strings', function () {
        ret = HandlebarHelper.concat('Hello ','World ','!!')
        assert.equal('Hello World !!', ret, 'concat string is "Hello World !!"');
    });
    it('should not throw error when 0 params are passed in', function () {
        ret = HandlebarHelper.concat()
        assert.equal('', ret, 'empty params');
    });
    it('should concatenate and return string when a non strings are passed in params', function () {
        ret = HandlebarHelper.concat('Hello ', 123, ' ', true)
        assert.equal('Hello 123 true', ret, 'concat string is "Hello 123 true"');
    });
```

**Comment by alewitt2** [comment]
@avnindersingh are you ok with the changes i have made? does this still work for your use case?

**Comment by avnindersingh** [comment]
> @avnindersingh are you ok with the changes i have made? does this still work for your use case?

@alewitt2 yes .. thanks

**Review by alewitt2 — APPROVED** [review]

**Pull request merged**